src/app/clusters/time-synchronization-server/time-synchronization-server.h

```diff
@@ -104,13 +89,28 @@ class TimeSynchronizationServer : public FabricTable::Delegate
     CHIP_ERROR SetUTCTime(chip::EndpointId ep, uint64_t utcTime, GranularityEnum granularity, TimeSourceEnum source);
     CHIP_ERROR GetLocalTime(chip::EndpointId ep, DataModel::Nullable<uint64_t> & localTime);
     GranularityEnum & GetGranularity() { return mGranularity; }
+    TimeSourceEnum & GetTimeSource() { return mTimeSource; }
 
     void ScheduleDelayedAction(System::Clock::Seconds32 delay, System::TimerCompleteCallback action, void * aAppState);
 
+    /**
+     * @brief Updates time zone to the correct item from the time zone list.
+     * If the TimeState has changed from the previous state, the function will emit a TimeZoneStatus event.
+     *
+     * Returns kInvalid if an error occurrs.
+     * Otherwise, it returns kActive since there is always a default time zone available.
+     * kStopped is not used in this function.
+     */
     TimeState UpdateTimeZoneState();
+    /**
+     * @brief Updates DST offset to the correct item from the DSTOffset list.
+     * If the DST offset has changed from the previous state or value, the function will emit a DSTStatus event.
+     *
+     * Returns kInvalid if an error occurrs or the DSTOffset list is empty.
+     * Returns kActive if a DST offset is being used.
+     * Returns kStopped if no DST offset is applicable at the current time.
```

Comment:
I would have expected kStopped (i.e. "permanent item") to correspond to empty list, but maybe I am missing something? Spec certainly seems to allow empty list, right?